engine/engine.go

```diff
@@ -22,4 +22,5 @@ type Engine interface {
 	Shutdown()
 
 	DumpInfo(output io.Writer) error
+	PublishWithJobID(namespace, queue, storedJobID string, body []byte, ttlSecond, delaySecond uint32, tries uint16) (jobID string, err error)
```

Comment:
Maybe `PublishJob(req JobRequest) (jobID string, err error)`?

Reply:
> Maybe `PublishJob(req JobRequest) (jobID string, err error)`?

yes, will change the API schema.

Reply:
I think this API should be `PublishJob(job engine.Job) (jobID string, error)`